packages/near-api-js/src/wallet-account.ts

```diff
@@ -102,7 +102,16 @@ export class WalletConnection {
             });
         }
         this._near = near;
-        const authDataKey = appKeyPrefix + LOCAL_STORAGE_KEY_SUFFIX;
+        const DEPRECATED_authDataKey = appKeyPrefix + LOCAL_STORAGE_KEY_SUFFIX;
+        const DEPRECATED_authData = JSON.parse(window.localStorage.getItem(DEPRECATED_authDataKey));
+        const authDataKey = (appKeyPrefix || near.config.contractName || 'default') + LOCAL_STORAGE_KEY_SUFFIX;
+
+        // Migrate dApps that previously used a faulty authDataKey.
+        if (DEPRECATED_authData) {
+            window.localStorage.setItem(authDataKey, JSON.stringify(DEPRECATED_authData));
+            window.localStorage.removeItem(DEPRECATED_authDataKey);
```

Comment:
Should we remove before adding the new data? I'm concerned about cases where `DEPRECATED_authDataKey` and `authDataKey` have the same value